app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt

```diff
@@ -0,0 +1,158 @@
+package org.commcare.connect.network.connectId.parser
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.commcare.CommCareTestApplication
+import org.commcare.android.database.connect.models.PersonalIdSessionData
+import org.json.JSONException
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+
+/**
+ * Unit tests for AddOrVerifyNameParser
+ */
+@Config(application = CommCareTestApplication::class)
+@RunWith(AndroidJUnit4::class)
+class AddOrVerifyNameParserTest {
+    private lateinit var parser: AddOrVerifyNameParser
+    private lateinit var sessionData: PersonalIdSessionData
+
+    @Before
+    fun setUp() {
+        parser = AddOrVerifyNameParser()
+        sessionData = PersonalIdSessionData()
+    }
+
+    @Test
+    fun testParseCompleteValidResponse() {
+        // Arrange
+        val json =
+            JSONObject().apply {
+                put("account_exists", true)
+                put(
+                    "photo",
+                    "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg" +
+                        "==",
+                )
+            }
+
+        // Act
+        parser.parse(json, sessionData)
+
+        // Assert
+        assertEquals(true, sessionData.accountExists)
+        assertEquals(
+            "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==",
+            sessionData.photoBase64,
+        )
+    }
```

Comment:
_**Optional: **_ We could extract the long string into a variable

```suggestion
    @Test
    fun testParseCompleteValidResponse() {
        // Arrange
        val photoBase64 =
            "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg=="
        val json =
            JSONObject().apply {
                put("account_exists", true)
                put("photo", photoBase64)
            }

        // Act
        parser.parse(json, sessionData)

        // Assert
        assertEquals(true, sessionData.accountExists)
        assertEquals(photoBase64, sessionData.photoBase64)
    }
```